src/constructors.jl

```diff
@@ -83,13 +153,16 @@ contents of the complex matrix `A`.
 """
 function LinearOperator(
   ::Type{T},
-  nrow::Int,
-  ncol::Int,
+  nrow::I,
+  ncol::I,
   symmetric::Bool,
   hermitian::Bool,
   prod,
   tprod = nothing,
   ctprod = nothing,
```

Comment:
These should probably change to `prod!` and similar.
I noticed here first, but there are a few places where the change is not made.